geode-core/src/main/java/org/apache/geode/cache/configuration/ClusterCacheElement.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.geode.cache.configuration;
+
+import org.apache.geode.annotations.Experimental;
+import org.apache.geode.cache.Cache;
+
+@Experimental
+public interface ClusterCacheElement extends CacheElement {
+  enum Operation {
```

Comment:
I see `ClusterCacheElement` being sent to CRUD methods on `ClusterManagementService` interface. But I don't understand why the data structure itself needs to carry its own operation?

Reply:
Depending on which type of ClusterCacheElement it is (i.e. a new region, a new async event queue, a new index, etc), it will go in a different place in the CacheConfig. Therefore, the add is not the same -- and each element must be able to define how to add itself.

Reply:
@Bill Not sure I fully understand your comment? We have ClusterCacheElement (called ClusterConfigElement now), which defines a method called addTo(CacheConfig existing) (that is now wrapped by a ConfigurationPersister instance). It is up to the individual subclass of ClusterConfigElement (like ClusterRegionConfig) to return a ConfigurationPersister that implements this method. The implementation contains the logic to persist the specific instance of ClusterConfigElement into the cluster configuration on the locator. Essentially giving us the polymorphism you are referring to. 

So I'm not sure I understand how what you're saying is different to what's happening? Would be great to sync up tomorrow to discuss this.

Reply:
+1 on the rename to `ClusterConfigElement`. That's more accurate.

As for my comments about `Operation`: if various subtypes of `CacheElement` do not need their own redundant operation (verbs) then feel free to disregard my suggestion.